### PR TITLE
Funky Weather Forecast

### PR DESCRIPTION
**Pull request by Izzibizz**
## Netlify link
(https://funkyweatherforecast.netlify.app/)

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  displayForecast(fourDaysFilter)
+  }
+  console.log(fourDaysFilter)
+console.log(filteredForecast)
+}
+
+
+//function to display all weather data for the upcoming days
+const displayForecast = (array) => {
+  array.forEach((day) => {
```
Comment (wwenzz) [inline script.js:97]: I would use arr instead of array as that's probably a reserved word.

File: script.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+getData('dakar')
+
+//Function to update HTML with forecast
+const updateHtml = (data) => {
+  const weather = data.list[0].weather[0].main
+  const city = data.city.name.toUpperCase()
+  const temp = Math.round(data.list[0].main.temp)
+
+  //sunrise and sunset
+  const rise = new Date(data.city.sunrise * 1000).toLocaleTimeString([], {
```
Comment (wwenzz) [inline script.js:42]: Nice function choice!

File: script.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+//Fetching the API and converting to Json
+const getData = (city) => {
+  fetch(URL(city))
+    .then((response) => {
+      return response.json()
+    })
+    .then((data) => {
+      updateHtml(data)
+    })
+    .catch((error) => console.log('error fetching data', error))
```
Comment (wwenzz) [inline script.js:29]: nice catch!

File: style.css [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  padding-left: 4em;
+  text-align: center;
+  font-weight: 600;
+}
+::placeholder {
+  color: rgb(0, 0, 0);
+  opacity: 1;
+}
+
+::-ms-input-placeholder {
```
Comment (wwenzz) [inline style.css:40]: I like how you manage this selector to control the placeholder styling for IE browser.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+//Function to handle the user's searched city 
+const handleSearch = () => {
+  const chosenCity = searchInput.value
+  getData(chosenCity)
+}
+
+//Eventlisteners
+searchButton.addEventListener('click', handleSearch)
+searchInput.addEventListener('keyup', function (event) {
+  if (event.key == 'Enter') {
```
Comment (wwenzz) [inline script.js:118]: what if the user doesn't input anything? could this also be a condition checked by if else statement?

File: script.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  <h3>${weather}</h3></div>
+  <div class="sunrise-sunset"><div class="up-n-down"><h4>Sunrise</h4>
+  <h4>Sunset</h4></div><div class="time"><h4>${rise}</h4><h4>${set}</h4>
+  </div></div>`
+  styleAfterWeather(weather, temp)
+  filterByTime(data)
+}
+
+//Function that styles the background and colors to fit weather
+const styleAfterWeather = (weather, temp) => {
```
Comment (wwenzz) [inline script.js:62]: Great styling function!

File: script.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  const filteredForecast = data.list.filter((item) => 
+  item.dt_txt.includes('12:00'))
+  const fourDaysFilter = filteredForecast.slice(1)
+  forecastContainer.innerHTML = ''
+  if (currentTime >= 13) {
+  displayForecast(filteredForecast)
+  } else {
+  displayForecast(fourDaysFilter)
+  }
+  console.log(fourDaysFilter)
```
Comment (wwenzz) [inline script.js:90]: would be better to remove those two console.log as these are mainly for development purposes.

**Review by wwenzz — COMMENTED** [review]
Your design is amazing 🤩!!! I like how you style the background based on the weather and it looks very cool. Your code also looks neat and clean - love it! Just one small suggestion - I didn't see you round any numbers to 1 decimal place, maybe this is something you could look into and challenge yourself to find a solution?

**Review by AntonellaMorittu — CHANGES_REQUESTED** [review]
Hi Izabel, good job finishing your assignment! 

Please pay attention to these basic requirements:

- The app should have: 4-day forecast You are displaying all 5 days when we requested 4 
- the temperature should be rounded to 1 decimal 
- the app should follow as close as possible the chosen design

Fix these things and you're good to go:) Keep up the awesome work 🚀

**Pull request closed without merging**